infrastructure/modules/onpremise-storage/mongodb/configmap.tf

```diff
@@ -15,6 +15,61 @@ db.createUser(
 db.sample.drop()
 
 EOF
+
+  init_replica_js = <<EOF
+
+rs.initiate({
+  _id :  "rs0",
+  members: [
+%{for i, service in kubernetes_service.mongodb~}
+    { _id:  ${i}, host:  "${service.metadata.0.name}.${service.metadata.0.namespace}:${service.spec.0.type == "NodePort" ? service.spec.0.port.0.node_port : service.spec.0.port.0.port}" },
+%{endfor~}
+  ]
+})
+
+EOF
+
+  mongo_start_sh = <<EOF
+
+# TODO: put this file at another place
+CLUSTER_KEY=/data/db/cluster.key
+
+if [ ! -e $CLUSTER_KEY ] ; then
+  cp /cluster/cluster.key $CLUSTER_KEY
+  chmod 400 $CLUSTER_KEY
+fi
+
+/usr/local/bin/docker-entrypoint.sh mongod \
```

Comment:
You might need need to put it as the last command with an exec, in order to ensure signals are properly propagated. The rs init "job" would be started in background with a sleep.

Reply:
ping?